tests/src/test/java/docs/javadsl/SerializationTest.java

```diff
@@ -0,0 +1,230 @@
+import akka.kafka.Subscriptions;
+import akka.kafka.javadsl.Consumer;
+import akka.kafka.javadsl.EmbeddedKafkaWithSchemaRegistryTest;
+import akka.kafka.javadsl.Producer;
+import akka.stream.*;
+import akka.stream.javadsl.Keep;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import akka.stream.testkit.javadsl.StreamTestKit;
+import akka.testkit.javadsl.TestKit;
+// #jackson-imports
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.core.JsonParseException;
+// #jackson-imports
+import docs.scaladsl.SampleAvroClass;
+// #imports
+import io.confluent.kafka.serializers.AbstractKafkaAvroSerDeConfig;
+import io.confluent.kafka.serializers.KafkaAvroDeserializer;
+import io.confluent.kafka.serializers.KafkaAvroSerializer;
+// #imports
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.producer.ProducerRecord;
+// #imports
+import org.apache.kafka.common.serialization.Deserializer;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.serialization.StringSerializer;
+// #imports
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.slf4j.bridge.SLF4JBridgeHandler;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.*;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public class SerializationTest extends EmbeddedKafkaWithSchemaRegistryTest {
+
+  private static final ActorSystem sys =
+      ActorSystem.create(SerializationTest.class.getSimpleName());
+  private static final Materializer mat = ActorMaterializer.create(sys);
+
+  private static final int kafkaPort = KafkaPorts.SerializationTest();
+  private static final int zooKeeperPort = KafkaPorts.SerializationTest() + 1;
+  private static final int schemaRegistryPort = KafkaPorts.SerializationTest() + 2;
+  private static final String schemaRegistryUrl = "http://localhost:" + schemaRegistryPort;
+  private static final Executor ec = Executors.newSingleThreadExecutor();
+
+  @Override
+  public ActorSystem system() {
+    return sys;
+  }
+
+  @Override
+  public String bootstrapServers() {
+    return "localhost:" + kafkaPort;
+  }
+
+  @BeforeClass
+  public static void beforeClass() {
+    // Schema registry uses Glassfish which uses java.util.logging
+    SLF4JBridgeHandler.removeHandlersForRootLogger();
+    SLF4JBridgeHandler.install();
+    startEmbeddedKafka(kafkaPort, zooKeeperPort, schemaRegistryPort, 1);
+  }
+
+  @Test
+  public void jacksonDeSer() throws Exception {
+    final String topic = createTopic(nextNumber(), 1, 1);
```

Comment:
This now can use `1` again as the last number in the topic is incremented using the same AtomicInteger across all Java tests.